src/lib/engines/test.engine.js

```diff
@@ -0,0 +1,75 @@
+const _ = require('lodash');
+
+const listeners = {};
+const subscriptions = {};
+
+/*
+ * This engine simply calls the locally saved function reference.
+ *
+ * You set what function to call with listen or subscribe.  You call functions with send or publish.
+ *
+ * Useful for testing bunnyhop APIs
+ */
+function TestEngine (pluginAPI) {
+  const log = pluginAPI.getPluginLogger();
+  const name = pluginAPI.getServiceName();
+  if (!listeners[name]) listeners[name] = {};
+
+  log.info('Using Test Engine');
+
+  return {
+    send:
+      async (routingKey, message, options = {}) => {
+        // Take the first listener, round robin them
+        const listenFn = listeners[name][routingKey].shift();
+        if (listenFn) {
+          log.info(`  [<] calling test listener for ${routingKey}`);
+          const result = listenFn({ content: message });
```

Comment:
The listener gets more than just "content" in the rabbit implementation. stuff like options and properties are also passed down

Reply:
Doesn't seem to  impact any of my tests though, if you know what else it needs can you explain / document it?